symphony-bdk-core/src/main/java/com/symphony/bdk/core/client/loadbalancing/RoundRobinLoadBalancingStrategy.java

```diff
@@ -0,0 +1,38 @@
+package com.symphony.bdk.core.client.loadbalancing;
+
+import com.symphony.bdk.core.config.model.BdkServerConfig;
+
+import org.apiguardian.api.API;
+
+import java.util.List;
+
+/**
+ * The {@link LoadBalancingStrategy} corresponding to the
+ * {@link com.symphony.bdk.core.config.model.BdkLoadBalancingMode#ROUND_ROBIN} mode.
+ */
+@API(status = API.Status.INTERNAL)
+public class RoundRobinLoadBalancingStrategy implements LoadBalancingStrategy {
+
+  private List<BdkServerConfig> nodes;
+  private int currentIndex;
+
+  /**
+   *
+   * @param nodes the list of nodes to be load balanced across in a round-robin way.
+   */
+  public RoundRobinLoadBalancingStrategy(List<BdkServerConfig> nodes) {
+    this.nodes = nodes;
```

Comment:
should we do a defensive copy of the list here to avoid it to be changed later? (ok this is being vey picky...)